cleanlab/rank.py

```diff
@@ -109,57 +123,335 @@ def get_self_confidence_for_each_label(
-      probabilities that x belongs to each possible class.
-      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
-      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+      Predicted-probabilities in the same format expected by the `score_label_quality()` method.
 
     Returns
     -------
-      normalized_margin : np.array (float)
-        Return a score (between 0 and 1) for each example of its likelihood of
-        being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
-        normalized_margin = prob_label - max_prob_not_label"""
+    label_quality_scores : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled.
+      normalized_margin = prob_label - max_prob_not_label
+
+    """
 
     self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
     max_prob_not_label = np.array(
         [max(np.delete(pred_probs[i], l, -1)) for i, l in enumerate(labels)]
     )
     label_quality_scores = (self_confidence - max_prob_not_label + 1) / 2
     return label_quality_scores
+
+
+def get_confidence_weighted_entropy_for_each_label(
+    labels: np.array, pred_probs: np.array
+) -> np.array:
+    """Returns the "confidence weighted entropy" label-quality score for each datapoint.
+
+    This is a function to compute label-quality scores for classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    "confidence weighted entropy" is the normalized entropy divided by "self-confidence".
+
+    Parameters
+    ----------
+    labels : np.array
+      Labels in the same format expected by the `score_label_quality()` method.
+
+    pred_probs : np.array (shape (N, K))
+      Predicted-probabilities in the same format expected by the `score_label_quality()` method.
+
+    Returns
+    -------
+    label_quality_scores : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled.
+
+    """
+
+    self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
+
+    # Divide entropy by self confidence
+    label_quality_scores = get_entropy(**{"pred_probs": pred_probs}) / self_confidence
+
+    # Rescale
+    label_quality_scores = np.log(label_quality_scores + 1) / label_quality_scores
+
+    return label_quality_scores
+
+
+def get_entropy(pred_probs: np.array) -> np.array:
+    """Returns the normalized entropy of pred_probs.
+
+    Read more about normalized entropy here: https://en.wikipedia.org/wiki/Entropy_(information_theory)
+
+    Normalized entropy is used in active learning for uncertainty sampling: https://towardsdatascience.com/uncertainty-sampling-cheatsheet-ec57bc067c0b
+
+    Normalized entropy is between 0 and 1. Higher values of entropy indicate higher uncertainty in the model's prediction of the correct label.
```

Comment:
Can we make this the 2nd sentence of the docstring before the external links are stated. I'd also add one more sentence: 

"Unlike label-quality scores, entropy only depends on the model's predictions, not the given label".

Reply:
Done, thanks
3dbbe7054aeae25b8bddd73a53838cce35b06fa9